src/Jobs/ChannelMonitorJob.cs

```diff
@@ -0,0 +1,163 @@
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+using FundsManager.Data;
+using FundsManager.Data.Models;
+using FundsManager.Data.Repositories.Interfaces;
+using FundsManager.Services;
+using Google.Protobuf;
+using Grpc.Core;
+using Lnrpc;
+using Microsoft.EntityFrameworkCore;
+using Quartz;
+using Channel = Lnrpc.Channel;
+
+namespace FundsManager.Jobs;
+
+/// <summary>
+/// Job for update the status of the channels
+/// </summary>
+/// <returns></returns>
+[DisallowConcurrentExecution]
+public class ChannelMonitorJob : IJob
+{
+    private readonly ILogger<ChannelMonitorJob> _logger;
+    private readonly IDbContextFactory<ApplicationDbContext> _dbContextFactory;
+    private readonly INodeRepository _nodeRepository;
+    private readonly ILightningService _lightningService;
+
+    public ChannelMonitorJob(ILogger<ChannelMonitorJob> logger, IDbContextFactory<ApplicationDbContext> dbContextFactory, INodeRepository nodeRepository, ILightningService lightningService)
+    {
+        _logger = logger;
+        _dbContextFactory = dbContextFactory;
+        _nodeRepository = nodeRepository;
+        _lightningService = lightningService;
+    }
+
+    public async Task Execute(IJobExecutionContext context)
+    {
+        var data = context.JobDetail.JobDataMap;
+        var nodeId = data.GetInt("nodeId");
+        _logger.LogInformation("Starting {JobName} for node {nodeId}... ", nameof(ChannelMonitorJob), nodeId);
+        try
+        {
+            var node1 = await _nodeRepository.GetById(nodeId);
+
+            if (node1 == null)
+            {
+                _logger.LogInformation("The node: {@Node} is no longer ready to be supported quartz jobs", node1);
+                return;
+            }
+
+            var client = LightningService.CreateLightningClient(node1.Endpoint);
+            var result = client.Execute(x => x.ListChannels(new ListChannelsRequest(),
+                new Metadata {
+                    {"macaroon", node1.ChannelAdminMacaroon}
+                }, null, default));
+
+            foreach (var channel in result?.Channels)
+            {
+                var node2 = await _nodeRepository.GetByPubkey(channel.RemotePubkey);
+
+                if (node2 == null)
+                {
+                    var foundNode = await _lightningService.GetNodeInfo(channel.RemotePubkey);
+                    if (foundNode == null)
+                    {
+                        throw new Exception("Node info not found");
+                    }
+
+                    node2 = new Node()
+                    {
+                        Name = foundNode.Alias,
+                        PubKey = foundNode.PubKey,
+                    };
+                    var addNode = await _nodeRepository.AddAsync(node2);
+                    if (!addNode.Item1)
+                    {
+                        throw new Exception(addNode.Item2);
+                    }
+                }
```

Comment:
I think this logic of creating a new node if not found in the repository should be extracted, maybe creating a method in the repository if makes sense that actually creates a new one passing a lightning client? Or at least extracting it outside of this function if it doesn't makes sense in that scope, what do you think?